src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+
+                            ElseIf currentNestingLevel = 0 Then
+                                ' field at the top level didn't match a tuple backing field, simply add.
+                                members.Add(New TupleFieldSymbol(Me, field.AsMember(currentUnderlying), -members.Count - 1))
+                            End If
+
+                        Case SymbolKind.NamedType
+                                        ' We are dropping nested types, if any. Pending real need.
+
+                        Case SymbolKind.Property
+                            If currentNestingLevel = 0 Then
+                                members.Add(New TuplePropertySymbol(Me, DirectCast(member, PropertySymbol).AsMember(currentUnderlying)))
+                            End If
+
+                        Case SymbolKind.Event
+                            If currentNestingLevel = 0 Then
+                                members.Add(New TupleEventSymbol(Me, DirectCast(member, EventSymbol).AsMember(currentUnderlying)))
+                            End If
+
+                        Case Else
+                            If currentNestingLevel = 0 Then
+                                Throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                            End If
+                    End Select
+                Next
+
+                If currentUnderlying.Arity <> RestPosition Then
+                    Exit Do
+                End If
+
+                Dim oldUnderlying = currentUnderlying
+                currentUnderlying = oldUnderlying.TypeArgumentsNoUseSiteDiagnostics(RestPosition - 1).TupleUnderlyingType
+                currentNestingLevel += 1
+
+                If currentUnderlying.Arity <> RestPosition Then
+                    ' refresh members And target fields
+                    underlyingMembers = currentUnderlying.OriginalDefinition.GetMembers()
+                    currentFieldsForElements.Clear()
+                    CollectTargetTupleFields(currentUnderlying, currentFieldsForElements)
+                Else
+                    Debug.Assert(oldUnderlying.OriginalDefinition Is currentUnderlying.OriginalDefinition)
+                End If
+            Loop
+
+            currentFieldsForElements.Free()
+
+            ' At the end, add remaining virtual fields
+            For i As Integer = 0 To namesOfVirtualFields.Count - 1
+                Dim Name As String = namesOfVirtualFields(i)
+                If Name IsNot Nothing Then
+                    ' We couldn't find a backing field for this vitual field. It will be an error to access it. 
+                    Dim fieldRemainder As Integer ' one - based
+                    Dim fieldChainLength = NumberOfValueTuples(i + 1, fieldRemainder)
+                    Dim container As NamedTypeSymbol = GetNestedTupleUnderlyingType(_underlyingType, fieldChainLength - 1).OriginalDefinition
+
+                    Dim DiagnosticInfo = If(container.IsErrorType(),
```

Comment:
done

---
In reply to: [71569767](https://github.com/dotnet/roslyn/pull/12596#discussion_r71569767) [](ancestors = 71569767)